scripts/validate-cloudtrail-logs.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+##? Usage: validate-cloudtrail-logs.sh [options] ACCOUNT START_FROM [UP_TO]
+##?
+##? Options:
+##?   --config=FILE  Path to file containing details of trails (default: $DM_CREDENTIALS_REPO/jenkins-vars/cloudtrail.yaml)
+##?   -v, --verbose
+##?   -n, --dry-run
+##?   -h, --help
+
+set -eo pipefail
+
+help=$(grep "^##?" "$0" | cut -c 5-)
+eval "$(docopts -h "$help" : "$@")"
+
+VERBOSE="${verbose:-$VERBOSE}"
+$VERBOSE && set -x
+
+[ -n "$config" ] && CONFIG_FILE="$config" || CONFIG_FILE="$DM_CREDENTIALS_REPO/jenkins-vars/cloudtrail.yaml"
+[ -f "$CONFIG_FILE" ] || { echo "error: config file '$CONFIG_FILE' does not exist"; exit 64; }
+
+[ -n "$UP_TO" ] || UP_TO="now"
+START_TIME="$(date --date="$START_FROM" --utc --iso-8601=minutes)" || exit 64
+END_TIME="$(date --date="$UP_TO" --utc --iso-8601=minutes)" || exit 64
+
+export ACCOUNT
+account="$("$DM_CREDENTIALS_REPO"/sops-wrapper -d --extract '["cloudtrail_validate_logs_roles"]' "$CONFIG_FILE" | yq 'map(select(.account == env.ACCOUNT))[0]')"
+
+AWS_PROFILE="$(echo "$account" | yq -r '.profile.name')"
+
+COMMAND="aws cloudtrail validate-logs"
+
+COMMAND+=" --trail-arn $(echo "$account" | yq -r '.trail_arn')"
+COMMAND+=" --start-time $START_TIME"
+[ -n "$END_TIME" ] && COMMAND+=" --end-time $END_TIME"
+
+if $VERBOSE
+then
+  COMMAND+=" --verbose"
+fi
+
+export AWS_PROFILE
+echo AWS_PROFILE="$AWS_PROFILE" $COMMAND
+$dry_run && exit
+$COMMAND
```

Comment:
Can we add a condition to also notify on in `$COMMAND | grep INVALID | wc -l`. If it is zero, we are good, if it is greater than zero echo the grep result to be posted to slack and exit with non-zero.

Reply:
Right, I foolishly assumed that the exit code would be non-zero if validation failed...

Reply:
You wouldn't happen to know if it's possible to cause a failed validation in someway?

Reply:
My bash skills are very rusty, but let me try:
`IS_INVALID = $($COMMAND | grep INVALID)`
`if $(echo $IS_INVALID  | wc -l) > 0`
`then $echo $IS_INVALID | exit 1`
`fi`

Reply:
The echo in then statement will give us the trail logs/digests that failed to validate, and the exit will trigger the error, which would send the alert to slack (if I understood the jenkins code correctly).

Reply:
I've made an attempt based on grep and tee, what do you think?